teos/src/watcher.rs

```diff
@@ -806,50 +860,87 @@ impl chain::Listen for Watcher {
                 self.filter_breaches(self.get_breaches(locator_tx_map));
 
             // Send data to the Responder
-            let mut appointments_to_delete = HashSet::from_iter(invalid_breaches.into_keys());
-            let mut delivered_appointments = HashSet::new();
-            for (uuid, breach) in valid_breaches {
-                log::info!(
-                    "Notifying Responder and deleting appointment (uuid: {})",
-                    uuid
-                );
+            let responder = self.responder.clone();
+            let appointments = self.appointments.clone();
+            let locator_uuid_map = self.locator_uuid_map.clone();
+            let gatekeeper = self.gatekeeper.clone();
+            let dbm = self.dbm.clone();
+            join_handle_option = Some(tokio::spawn(async move {
+                let mut appointments_to_delete = HashSet::from_iter(
+                    invalid_breaches.into_keys());
+                let mut delivered_appointments = HashSet::new();
```

Comment:
Note: This would make this method exit early and a task will be spawned on tokio to do the job. This could potentially make the responder's/gatekeeper's own `block_connected` method run before or interleaving with this task.